### DISCUSSION TITLE
5250 emulation in vscode

### DISCUSSION THREAD

**Discussion by szsascha** [codefori/vscode-ibmi#229] [Ideas]
This discussion is a bit offtopic to vscode for ibmi but I think this is a good place for it. I hope you don't mind. If it does, please just close it. :)

I'm wondering if a 5250 emulation in js / vscode could be useful. Maybe as a feature of vscode for ibmi, as an independent extension or completely without vscode. For example tn5250j is really useful to integrate it in other java projects in my opinion. But unfortunately no maintainer cares about the project anymore.

What is your opinion regarding a 5250j emulation in vscode?

**Comment by worksofliam** [comment]
It would be awesome to be able to have a tab which actually is a 5250 emulator.

The issue is actually getting the emulator there. I personally will not allow any Java to be integrated into this extension. I like the fact that installing this extension doesn't require any other software to be installed.

Why isn't there a JavaScript emulator in the browser yet that doesn't require any 3rd party softwares? :o

**Reply by szsascha** [reply]
Thanks for your reply!

I'm with you. A piece of Java code will not fit in this project. The 5250 lowlevel stuff is pretty exciting in my opinion. So I could imagine to create a 5250 client in JS/TS. But it's too much work to do this alone in freetime.

There is also one little issue. 5250 is based on TCP/IP so you have to use sockets. You cannot use native sockets in browser. But it should work in node.
